src/ucp/wireup/wireup_cm.c

```diff
@@ -765,23 +760,44 @@ ucp_ep_cm_server_create_connected(ucp_worker_h worker, unsigned ep_init_flags,
                                           ep_init_flags,
                                           "conn_request on uct_listener", &ep);
     if (status != UCS_OK) {
-        return status;
+        ucs_warn("server ep %p failed to connect to worker address on device %s, tl_bitmap 0x%zx, status %s",
+                 ep, conn_request->dev_name, tl_bitmap,
+                 ucs_status_string(status));
+        uct_listener_reject(conn_request->uct.listener, conn_request->uct_req);
+        goto out;
     }
 
     status = ucp_wireup_connect_local(ep, remote_addr, NULL);
     if (status != UCS_OK) {
-        return status;
+        ucs_warn("server ep %p failed to connect to remote address on device %s, tl_bitmap 0x%zx, status %s",
+                 ep, conn_request->dev_name, tl_bitmap,
+                 ucs_status_string(status));
+        uct_listener_reject(conn_request->uct.listener, conn_request->uct_req);
+        ucp_ep_destroy_internal(ep);
+        goto out;
     }
 
-    status = ucp_ep_cm_connect_server_lane(ep, conn_request);
+    status = ucp_ep_cm_connect_server_lane(ep, conn_request->uct.listener,
+                                           conn_request->uct_req);
     if (status != UCS_OK) {
-        return status;
+        ucs_warn("server ep %p failed to connect CM lane on device %s, tl_bitmap 0x%zx, status %s",
+                 ep, conn_request->dev_name, tl_bitmap,
+                 ucs_status_string(status));
+        ucp_ep_destroy_internal(ep);
```

Comment:
no reject here?

Reply:
`ucp_ep_cm_connect_server_lane` does reject on failure because it does `uct_ep_create` on success

Reply:
how it's related? if we failed to create ep it's better to reject, no? since we have reject in line 775 for other error during connect

Reply:
> how it's related? if we failed to create ep it's better to reject, no? since we have reject in line 775 for other error during connect

If we failed to create EP on server side, we destroy `event->id` and ack event:
https://github.com/openucx/ucx/blob/fe2b77cb249e3d508efb6a052c2dc5b8a6af2fa5/src/uct/ib/rdmacm/rdmacm_cm_ep.c#L398
so we can't reject this event/id after this

@alinask I can't find a mention about UCT `conn_request` has to be released by calling `uct_ep_create` or `uct_listener_reject`. Did not we lost it during refactoring?

Reply:
@evgeny-leksikov looks like we only mention it for UCP in ucp_listener_conn_callback_t doc.
https://github.com/openucx/ucx/blob/master/src/ucp/api/ucp_def.h#L384

Reply:
@evgeny-leksikov ok, so this is correct, need to update doc for uct_ep_create, or uct_ep_params.conn_request - that even in case of failure, the conn_request is released

Reply:
fixed in separate PR https://github.com/openucx/ucx/pull/5311